### DISCUSSION TITLE
[Run issue]: TsIRTInterPulse: The number of generated free radicals < 0.

### DISCUSSION THREAD

**Discussion by Vacuoler** [topas-nbio/TOPAS-nBio#55] [Run-Problem]
### Contact Details

sunjianhan@stu.pku.edu.cn

### What was the issue?

Respected expert,

Hello!

When I used TsIRTInterPulse to simulate the free radicals generated by pulsed beams, the number of generated free radicals was less than 0 over time, just like the data below. The chemical reactions I used are the default  PlanteReactions.txt  and several type VI reactions added by Jose Ramos Mendez in problem IRTcummulative Example #7, PlanteTypeVI.txt. I don't know where the problem is. 

<img width="581" alt="image" src="https://github.com/user-attachments/assets/d48fdc15-fda7-40e6-8e98-b7ed2217cb16">

My code and the cited chemical reactions are listed as follows:

[2MeV_1ps_1Hz_1pulses_10Gy.txt](https://github.com/user-attachments/files/18004056/2MeV_1ps_1Hz_1pulses_10Gy.txt)
[PlanteReactions.txt](https://github.com/user-attachments/files/18004058/PlanteReactions.txt)
[PlanteTypeVI.txt](https://github.com/user-attachments/files/18004062/PlanteTypeVI.txt)

#######################
#######################

i:Ts/Verbosity = 1
i:Ts/ShowHistoryCountAtInterval = 260
Ts/NumberOfThreads = 1
b:Ts/BufferThreadOutput = "True"
b:Ts/ShowCPUTime = "True"

i:Ts/Seed = 1234

#######################
# Physics & Chemistry #
#######################

#sv:Ph/Default/Modules = 4 "TsEmDNAPhysics" "TsEmDNAChemistry"  "g4em-dna_opt2" "g4em-dna-chemistry"
sv:Ph/Default/Modules = 2 "g4em-dna_opt2" "TsEmDNAChemistry"

s:Ph/Default/Electron/SetElasticScatteringModel   = "CPA100"
s:Ph/Default/Electron/SetExcitationModel          = "Emfietzoglou"
s:Ph/Default/Electron/SetIonisationModel          = "Emfietzoglou"
b:Ph/Default/Electron/ActiveVibExcitation         = "True"
b:Ph/Default/Electron/ActiveAttachment            = "True"
s:Ch/ChemistryName = "Plante"
includeFile        = PlanteReactions.txt
includeFile        = PlanteTypeVI.txt
b:Sc/gvalue/EnableHighTimeScavengers = "True"

b:Ch/Plante/ChemicalStageTransportActive = "True"
d:Ch/Plante/ChemicalStageTimeEnd         = 200e6 ps
#dv:Ch/Plante/ChemicalStageTimeStepsHighEdges = 1 200e6 ps
#dv:Ch/Plante/ChemicalStageTimeStepsResolutions = 1 2 ps

#######################
#######################

d:Ge/World/HLX= 10 um 
d:Ge/World/HLY= 10 um 
d:Ge/World/HLZ= 10 um 
s:Ge/World/Material ="G4_WATER"

s:Ge/target/Parent   = "World"
s:Ge/target/Material = "G4_WATER"
s:Ge/target/Type     = "TsBox"
d:Ge/target/HLX      = 10 um
d:Ge/target/HLY      = 10 um
d:Ge/target/HLZ      = 10 um

#######################
#######################

s:So/Example/Type                     = "Volumetric"
s:So/Example/ActiveMaterial           = "G4_WATER"
s:So/Example/Component                = "target" 
s:So/Example/BeamParticle             = "e-"
d:So/Example/BeamEnergy               = 2 MeV 
u:So/Example/BeamEnergySpread         = 0
ic:So/Example/NumberOfHistoriesInRun  = 1000000

#######################
#######################

s:Sc/gvalue/Quantity     = "TsIRTInterPulse" 
s:Sc/gvalue/Component    = "target"
s:Sc/gvalue/OutputFile   = "2MeV_100ps_1Hz_1pulses_10Gy"
s:Sc/gvalue/IfOutputFileAlreadyExists = "Overwrite"
s:Sc/gvalue/OutputType   = "ascii"
i:Sc/gvalue/TimeBins     = 500
d:Sc/gvalue/TimeLower    = 1.0e0 ps
d:Sc/gvalue/TimeUpper    = 10 us 
i:Sc/gvalue/NumberOfPulses = 1
d:Sc/gvalue/PulsesFrequency = 1e0 /s
d:Sc/gvalue/PrescribedDose    = 10 Gy 
s:Sc/gvalue/PulseDistribution = "Gaussian"
d:Sc/gvalue/PulseTimeMean     = 50 ps 
d:Sc/gvalue/PulseTimeFWHM     = 100 ps 
sv:Sc/gvalue/ReportMoleculesNamed = 10 "OH^0" "e_aq^-1" "H3O^1" "H2O2^0" "H^0" "H_2^0" "OH^-1" "O2^-1" "HO2^0" "O2^0"
s:Sc/gvalue/SensitiveVolumeName  = "target"
b:Sc/gvalue/ForceLowTimeCutTo1ps = "True"

#######################

Thank you very much.

### TOPAS-nBio category

Chemistry

### Version

3.0

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Attestation

- [X] I testify that I have fully read the bug reporting guideline beforehand.

**Comment by tmasilela** [comment]
Hi,

Are you sure you're using the latest version of TOPAS (OpenTOPAS v4.0.0) and TOPAS-nBio (v3.0)? This is a known issue that I posted about in the previous forum but should be fixed in the latest version of the software.

Hope that fixes your issue.

All the best,

Thongchai

**Reply by Vacuoler** [reply]
Excuse me, how can I install TOPAS nbio v3.0 separately? I am unable to determine whether the version I installed is this one. I downloaded topas-nbio on September 1, 2024.

Thank you very much.
